File: backend/src/main/java/kz/hackquest/model/Lab.java
package kz.hackquest.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "labs")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Lab {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    private String category;

    @Column(length = 2000)
    private String description;

    private String difficulty;

    private Integer duration;

    private Integer xpReward;

    @Column(length = 5000)
    private String briefing;

    @Column(length = 2000)
    private String hint;

    @Column(length = 5000)
    private String vulnerableCode;

    @Column(length = 2000)
    private String correctPayload;

    private Boolean enabled;

    @PrePersist
    public void onCreate() {
        if (enabled == null) enabled = true;
    }
}

File: frontend/src/app/types/Lab.ts
export interface Lab {
  id: number;
  title: string;
  category: string;
  description: string;
  difficulty: "Новичок" | "Средний" | "Сложный" | "Эксперт";
  xpReward: number;
  duration: number;
  completed: boolean;
  unlocked: boolean;
  briefing: string;
  hint: string;
  vulnerableCode: string;
  correctPayload: string;
  targetInfo?: string;
}